container/lxd/container.go

```diff
@@ -39,15 +41,24 @@ type ContainerSpec struct {
 // Note that we pass these through as supplied. If an instance type constraint
 // has been specified along with specific cores/mem constraints,
 // LXD behaviour is to override with the specific ones even when lower.
-func (c *ContainerSpec) ApplyConstraints(cons constraints.Value) {
+func (c *ContainerSpec) ApplyConstraints(serverVersion string, cons constraints.Value) {
 	if cons.HasInstanceType() {
 		c.InstanceType = *cons.InstanceType
 	}
 	if cons.HasCpuCores() {
 		c.Config["limits.cpu"] = fmt.Sprintf("%d", *cons.CpuCores)
 	}
 	if cons.HasMem() {
-		c.Config["limits.memory"] = fmt.Sprintf("%dMiB", *cons.Mem)
+		// Ensure that for LXD versions 3.10+, we use the correct "MiB" suffix.
+		template := "%dMB"
+		if current, err := version.Parse(serverVersion); err == nil {
+			if min, err := version.NewDottedVersion("3.10"); err == nil {
```

Comment:
is this the easiest way to get a Version{3,10} ? it seems a little odd to parse on every call something that is essentially a const.

Reply:
Good call. Moved it to an explicitly created var. We don't actually need to parse and check for error.